rust/src/shards/gui/widgets/link.rs

```diff
@@ -0,0 +1,194 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright © 2022 Fragcolor Pte. Ltd. */
+
+use super::Link;
```

Comment:
Maybe we can just have link and simplify it to work for either url's if you pass :Url or otherwise by passing :Action
The only difference in behavior is what happens when clicking

Reply:
Hyperlink and Link have other differences:

|  | Input type | Output type |
|---|---|---|
| Hyperlink | String | String (input passthrough) |
| Link | Any | bool (whether clicked this frame)l |

What happens if both `:Url` and `:Action` are set: should we open the link AND execute the action?